src/ccmain/tesseractclass.cpp

```diff
@@ -79,6 +79,30 @@ Tesseract::Tesseract()
                  "Thresholding "
                  "method: 0 = Otsu, 1 = Adaptive Otsu, 2 = Sauvola",
                  this->params())
+    , INT_MEMBER(thresholding_sauvola_window_half_width, 25,
+                 "Window half-width for measuring local statistics",
+                 this->params())
+    , double_MEMBER(thresholding_sauvola_kfactor, 0.40,
+                 "Factor for reducing threshold due to variance. Must be 0 or positive number",
+                 this->params())
+    , INT_MEMBER(thresholding_adaptive_otsu_tile_x, 300,
+                 "Desired tile width. Actual size may vary. Must be >= 16",
+                 this->params())
+    , INT_MEMBER(thresholding_adaptive_otsu_tile_y, 300,
+                 "Desired tile height. Actual size may vary. Must be >= 16",
+                 this->params())
```

Comment:
As I said, it's not a good encapsulation of complexity for the user to think about absolute tile sizes. This depends so much on DPI and on RAM size. Rather, it should be the number of tiles which the full image is split into (`thresholding_num_tiles`?) or the relative tile size (`thresholding_tile_size_inch`?).

Reply:
>As I said, it's not a good encapsulation of complexity for the user to think about absolute tile sizes.

But that's what other implementations do.

I want to keep the tiles size and window size as absolute values, like the other implementations do.

Nevertheless, I plan to provide an option to turn on a bool parameter to scale these absolute numbers depending on the dpi info which is embedded in the image (or user provided dpi that Tesseract already supports).

With this option turn on, the absolute values will be related to the base dpi of 300x300. If the actual dpi is 600x600 the tiles and the window will be scaled up.

Reply:
> But that's what other implementations do.
> 
> I want to keep the tiles size and window size as absolute values, like the other implementations do.

It would help naming a few of those implementations. Normally I would agree and see that as a strong argument, but in this case I believe we are comparing implementations of different tasks – binarization or other low-level tools on the one hand, and robust layout analysis or other high-level tools on the other.

> Nevertheless, I plan to provide an option to turn on a bool parameter to scale these absolute numbers depending on the dpi info which is embedded in the image (or user provided dpi that Tesseract already supports).
> 
> With this option turned on, the absolute values will be related to the base dpi of 300x300. If the actual dpi is 600x600 the tiles and the window will be scaled up.

That would be a good compromise, the crux would be to make the parameter names and their description readily understandable.